src/library_html5.js

```diff
@@ -1119,7 +1119,10 @@ var LibraryHTML5 = {
   $fillFullscreenChangeEventData: function(eventStruct) {
     var fullscreenElement = document.fullscreenElement || document.mozFullScreenElement || document.webkitFullscreenElement || document.msFullscreenElement;
     var isFullscreen = !!fullscreenElement;
+#if !SAFE_HEAP
+    // Assigning a boolean to HEAP32 with expected type coercion.
     /** @suppress{checkTypes} */
+#endif
```

Comment:
Hmm, so this fails in SAFE_HEAP somehow, as an unneeded annotation..? I'm surprised closure doesn't just silently ignore it...

Reply:
Yeah me too..  it reports `[JSC_MISPLACED_SUPPRESS] @suppress annotation not allowed here. See https://github.com/google/closure-compiler/wiki/@suppress-annotations`

See the current waterfall failures: https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket/8822058280191013073/+/u/Emscripten_testsuite__SAFE_HEAP_/stdout

Reply:
Is there an "ignore JSC_MISPLACED_SUPPRESS" option perhaps..? :smile: 

I'm not opposed to this, but it does add odd clutter... I wonder if there is another way. But I guess the problem is that in safe-heap mode there is no assignment here (there is a function call), so nothing to check types on...

Could we add this annotation in `makeSetValue` (when not safe-heap)? Or would that apply too often? I tend to think it should be ok there.

Reply:
Yes.

I'd rather not universally suppress `JSC_MISPLACED_SUPPRESS` since it is generally useful.

Reply:
Another option is to avoid `makeSetValue` here. SAFE_HEAP will actually instrument the code using an acorn pass anyhow. However, there might be other modes that do want special handling... the only thing I can think of is memory64 mode in which i64 operations are different.

Reply:
I'd be ok with modifying `makeSetValue` as long as it was given enough information (another argument parhaps), but I wouldn't want to do it universally since in most cases the suppression is not needed.

Perhaps a better long term fix would be to upstream a change to closure to allow booleans to be assigned to arraybuffer slots?  Its doesn't seem unreasonable to me.

Reply:
> Another option is to avoid `makeSetValue` here. SAFE_HEAP will actually instrument the code using an acorn pass anyhow. However, there might be other modes that do want special handling... the only thing I can think of is memory64 mode in which i64 operations are different.

I'm not sure what you mean here..  `makeSetValue` is no the problem, its the assignment of a known boolean value of HEAPXX slot.. (which is what `makeSetValue` generates) which is the problem.

The SAFE_HEAP pass runs before closure so closure will either see a direct assignment of a function call depending on whether SAFE_HEAP is enabled.

Reply:
Oh, yes, the safe-heap pass runs before closure... good point.

Long-term fix of updating closure sounds best. This lgtm to land for now with a TODO